climpred/classes.py

```diff
@@ -90,13 +89,63 @@ def __init__(self, xobj):
             # makes applying prediction functions easier, etc.
             xobj = xobj.to_dataset()
         has_dims(xobj, ['init', 'lead'], 'PredictionEnsemble')
-        self.initialized = xobj
-        self.uninitialized = {}
+        # RXB:
+        # Will store all datasets in a large dictionary to make things like __getattr__
+        # easy, i.e., we can loop through everything at once. We can reserve certain
+        # names ('initialized', 'uninitialized', etc.) for special cases for applying
+        # prediction functions to.
+        #
+        # Perhaps a future solution is to have a nested dictionary so you can have
+        # multiple prediction ensembles under the 'initialized' sub-dictionary.
+        self._datasets = {'initialized': xobj}
 
     # when you just print it interactively
     # https://stackoverflow.com/questions/1535327/how-to-print-objects-of-class-using-print
     def __repr__(self):
-        return _display_metadata(self)
+        # REINSTATE THIS.
+        return 'Printing is temporarily disabled.'
+        # return _display_metadata(self)
+
+    def __getattr__(self, name):
+        """Allows for xarray methods to be applied to our prediction objects.
+
+        Args:
+            * name: Function, e.g., .isel() or .sum().
+        """
+        # Temporarily registers attribute with the object.
+        setattr(self, name, self)
+
+        def wrapper(*args, **kwargs):
+            """Applies arbitrary function to all datasets in the PredictionEnsemble
+            object.
+
+            Got this from: https://stackoverflow.com/questions/41919499/
+            how-to-call-undefined-methods-sequentially-in-python-class
+            """
+            # Create temporary copy to modify to avoid inplace operation.
+            datasets = self._datasets.copy()
+            datasets = {
+                k: getattr(v, name)(*args, **kwargs)
+                for (k, v) in self._datasets.items()
+            }
+            # Instantiates new object with the modified datasets.
+            return self._construct_direct(datasets)
+
+        # Remove rergistered attribute.
+        delattr(self, name)
```

Comment:
Is `setattr` and `delattr` needed for any reason? This also works without either of them. I used `setattr` originally based on @ahuang11's code snippet. I got rid of the public attributes in our classes. Right now we only have `cp._datasets` so we shouldn't expect needing to overwrite or clash into someone using an xarray method that's the same name as an attribute.

Reply:
It's been a while, I honestly don't remember ha, but if it works without it, then I encourage deleting it to lessen the clutter and confusion.